sdk/azidentity/azidentity_test.go

```diff
@@ -1001,6 +1006,108 @@ func TestTokenCachePersistenceOptions(t *testing.T) {
 	}
 }
 
+func TestDoForClient(t *testing.T) {
+	var (
+		policyHeaderName  = "PolicyHeader"
+		policyHeaderValue = "policyvalue"
+
+		reqBody  = []byte(`{"request": "azidentity"}`)
+		respBody = []byte(`{"response": "golang"}`)
+	)
+
+	tests := map[string]struct {
+		method  string
+		path    string
+		body    io.Reader
+		headers http.Header
+
+		wantErr bool
+	}{
+		"happy path": {
+			method: http.MethodGet,
+			path:   "/foo/bar",
+			body:   bytes.NewBuffer(reqBody),
+			headers: http.Header{
+				"Header": []string{"value1", "value2"},
+			},
+			wantErr: false,
+		},
+		"no body": {
+			method:  http.MethodGet,
+			path:    "/",
+			body:    http.NoBody,
+			wantErr: false,
+		},
+	}
+
+	client, err := azcore.NewClient(module, version, azruntime.PipelineOptions{
+		// add PerCall policy to ensure doForClient calls .Pipeline.Do()
+		PerCall: []policy.Policy{
+			policyFunc(func(req *policy.Request) (*http.Response, error) {
+				req.Raw().Header.Set(policyHeaderName, policyHeaderValue)
+				return req.Next()
+			}),
+		},
+	}, nil)
+	require.NoError(t, err)
+
+	for name, tt := range tests {
+		tt := tt
```

Comment:
If someone writes a bug like that, wouldn't you rather break the test to make it obvious?

Reply:
honestly? no, I would not. 

the only remedy to a captured loop variable is to add something like `tt := tt` before the goroutine.. in my opinion, what we're doing here essentially is taking the defensive approach to prevent unnecessary debugging efforts .. 

I mean in such simple test like this, it's probably going to be pretty clear what's happening, but if there's 100s lines of convoluted test, one might easily miss that they are capturing a loop variable when adding some functionality to the test, causing a bug that takes hours to debug only to learn they need to copy first ..

that's why I said this just became a habit to me.. it's a one liner that has no side effects and saved us a lot of headaches in my previous company, but if you're strongly against it, I'm going to remove it

Reply:
*Strongly* against it, I dunno, but certainly against it ☺️. I might look at this differently in product code where preventing weird behavior in production is critical, but these are tests. If someone wrote a test that would fail, or cause another test to fail, without this shadowing--which I think is extraordinarily unlikely--I would consider that a bug concealed by the shadowing, and I would prefer the tests to make the bug visible by failing.

Reply:
removed

Reply:
well.. 

1. `go vet` checks failed because `loop variable tt captured by func literal` 😁 
1. `go1.18` tests failed (newer go versions did not 🤔), and I wouldn't be surprised if it's because of not copying the variable first

I believe the variable copy is correct in this case. What do you suggest?

Reply:
`go vet` flags your test but not other similar tests because its subtests run in parallel, which is also why your test sometimes fails--calling `t.Parallel()` creates a race. Copying the variable prevents that race but I suggest just running the tests serially because they don't run notably faster in parallel.

Reply:
removed `t.Parallel()`